src/cc/bcc_proc.c

```diff
@@ -392,3 +392,74 @@ char *bcc_procutils_which_so(const char *libname, int pid) {
 void bcc_procutils_free(const char *ptr) {
   free((void *)ptr);
 }
+
+/* Detects the following languages + C. */
+const char *languages[] = {"java", "python", "ruby", "php", "node"};
+const int nb_languages = 5;
+
+char *bcc_procutils_language(int pid) {
+  char procfilename[22], line[4096], pathname[32], *str;
+  FILE *procfile;
+  int i, ret;
+
+  /* Look for clues in the process' command line. */
```

Comment:
Probably not with the languages we currently recognize. BTW, the logic should be a bit more flexible. For example, `irb` (the Ruby REPL) will not be recognized as Ruby by this logic, right?